.github/workflows/ci-generate.yaml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  pull_request:
+     branches:
+       - main
+
+jobs:
+  check-drift:
+    runs-on: ubuntu-latest
+    env:
+      REPO: ${{ github.event.repository.name }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-generate
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-generate
+
+      - name: Retain a copy of every generated file
+        run: |
+          for template_file in $(find . -type f -name '*.template'); do
+            generated_file="${template_file%.template}"
+            if [[ -f "{$generated_file}" ]]; then
+              # Keep a copy to diff with the output of the next step.
+              cp "${generated_file}" "${generated_file}.check"
+            fi
+          done
+
+      - name: Generate from templates
+        run: |
+          # Note: we use cargo-generate itself here as our wrapping of the cargo-generate library
+          # does not allow us to do useful things like --allow-commands and use local templates.
+          cargo generate --path . --name '${{ env.REPO }}' --allow-commands
+
+      - name: Check for drift between templates and generated files
+        run: |
+          for check_file in $(find '${{ env.REPO }}' -type f -name '*.check'); do
+            generated_file="${check_file%.check}"
+            if ! diff "${check_file}" "${generated_file}"; then
+              echo "File ${generated_file} has drift from ${generated_file}.template:"
+              echo "::error file=${generated_file}::Template drift detected."
+              exit 1
```

Comment:
This `exit` should be moved to after the `for` loop so that all errors can be printed first.